contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,691 @@
+        require(orders[_orderID].buyer == _msgSender(), "OTCWrapper: only buyer can undo the order");
+
+        orderStatus[_orderID] = OrderStatus.Failed;
+
+        emit OrderCancelled(_orderID);
+    }
+
+    /**
+     * @notice executes an order
+     * @dev can only be called by whitelisted market makers
+     *      requires that product and collateral have already been whitelisted beforehand
+     *      ensure collateral naked cap from Controller.sol is high enough for the additional collateral
+     *      ensure setUpperBoundValues and setSpotShock from MarginCalculator.sol have been set up
+     * @param _orderID id of the order
+     * @param _userSignature user permit signature
+     * @param _mmSignature market maker permit signature
+     * @param _premium order premium amount
+     * @param _collateralAsset collateral asset address
+     * @param _collateralAmount collateral amount
+     */
+    function executeOrder(
+        uint256 _orderID,
+        Permit calldata _userSignature,
+        Permit calldata _mmSignature,
+        uint256 _premium,
+        address _collateralAsset,
+        uint256 _collateralAmount
+    ) external nonReentrant {
+        require(orderStatus[_orderID] == OrderStatus.Pending, "OTCWrapper: inexistent or unsuccessful order");
+        require(isWhitelistedMarketMaker[_msgSender()], "OTCWrapper: address not whitelisted marketmaker");
+        require(_userSignature.amount >= _premium, "OTCWrapper: insufficient amount");
+
+        Order memory order = orders[_orderID];
+
+        require(_userSignature.acct == order.buyer, "OTCWrapper: signer is not the buyer");
+        require(block.timestamp <= order.openedAt.add(fillDeadline), "OTCWrapper: deadline has passed");
+        require(whitelist.isWhitelistedCollateral(_collateralAsset), "OTCWrapper: collateral is not whitelisted");
+
+        require(
+            marginRequirements.checkMintCollateral(
+                _msgSender(),
+                order.notional,
+                order.underlying,
+                order.isPut,
+                _collateralAmount,
+                _collateralAsset
+            ),
+            "OTCWrapper: insufficient collateral"
+        );
+
+        // user inflow
+        _deposit(
+            _userSignature.acct,
+            USDC,
+            _premium,
+            _userSignature.deadline,
+            _userSignature.v,
+            _userSignature.r,
+            _userSignature.s
+        );
+
+        // market maker inflow
+        _deposit(
+            _mmSignature.acct,
+            _collateralAsset,
+            _collateralAmount,
+            _mmSignature.deadline,
+            _mmSignature.v,
+            _mmSignature.r,
+            _mmSignature.s
+        );
+
+        uint256 orderFee = (order.notional.mul(fee[order.underlying])).div(10000); // eg. fee = 4bps = 0.04% , then need to divide by 100 again so (( 4 / 100 ) / 100)
```

Comment:
lets move the comment above the line and separate into 2 lines

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/commit/9ff2d61f014d96dcae3f2327e91c92d32d02dfaa